pkg/azdiskschedulerextender/main.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"encoding/json"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+	schedulerapi "k8s.io/kube-scheduler/extender/v1"
+	"time"
+)
+
+func init() {
+	klog.InitFlags(nil)
+}
+
+var (
+	metricsAddress = flag.String("metrics-address", "0.0.0.0:29604", "export the metrics")
+	azDiskSchedulerExtenderPort = flag.String("az-disk-scheduler-extender-port", "8080", "port used by az scheduler extender")	
```

Comment:
> `"az-disk-scheduler-extender-port"`

I believe we could simply use `"port"` here since the executable is already called "azdiskschedulerextender". If we combine this later into the azurediskplugin executable, we can use a more specific flag (e.g. `"scheduler-extender-port"`) then.

Reply:
It felt less intuitive to me, given we have at least one more port defined in the yaml. I have changed it to "port" for now, we can change it back as needed.